### PR TITLE
Use FeatureEntry and Stage for stars and notifiers

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
This change updates the notifiers logic to use the new schema entities. This change required a large refactor to how we handle detection of changed values when making edits and notifying users, so the bulk of the change resides in this refactor of feature editing (guide.py).

Explanation of changes by file:
- `internals/core_models.py`: Remove `stash_values()` and built-in notifier in the `Feature.put()` method (stash_values is no longer needed due to keeping track of changed values in guide.py).
- `internals/notifier.py`: Replace most uses of Feature with FeatureEntry, and use Stage entities as needed for fields (e.g. "shipped_milestone").
- `internals/notifier_helpers.py`: Finalize implementation to notify subscribers of changes using FeatureEntry and Stage entities. Now, a list of changes are passed in rather than determined during notification by checking Feature field values (this was no longer viable due to some fields now being moved to Stage entities).
- `internals/notifier_helpers_test.py`: These tests have been moved from review_models.py due to changes making the tests more aptly test the helper functions.
- `pages/guide.py`: A large refactoring of how fields are checked as being touched and updated in the Feature/FeatureEntry/Stage entities. Now, a number of constants are used to help simplify this process and declutter the post request method. Additionally, feature changes are tracked here, which make it easier to find which fields have been changed for activity logging and subscriber notifications.

### CODE REVIEW

File: internals/notifier_helpers.py [hunk 1]
```diff
@@ -13,32 +13,28 @@
...
         continue
       amendments.append(
-          review_models.Amendment(field_name=prop_name,
+          review_models.Amendment(field_name=field,
           old_value=str(old_val), new_value=str(new_val)))
   return amendments
 
-def notify_feature_subscribers_of_changes(feature,
-    amendments: list[review_models.Amendment], is_update: bool) -> None:
+def notify_feature_subscribers_of_changes(fe,
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply internals/notifier_helpers.py]: I originally did not know how to supply this type hint without causing a circular import, but I found [this info](https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/).

File: internals/notifier.py [hunk 2]
```diff
@@ -119,24 +124,26 @@ def convert_reasons_to_task(
...
   # configurable through some kind of user preference.
   changed_field_names = {c['prop_name'] for c in changes}
   results: dict[str, list[str]] = {}
-
+  stage_type = core_enums.STAGE_TYPES_SHIPPING[fe.feature_type] or 0
+  ship_stage = fe_stages[stage_type]
   # Check if feature has some other milestone set, but not webview.
-  if (feature.shipped_android_milestone and
-      not feature.shipped_webview_milestone):
+  if (ship_stage.milestones.android_first and
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply internals/notifier.py:138]: Very true - updated!

File: pages/guide.py [hunk 3]
```diff
@@ -123,9 +124,97 @@ def write_gates_and_stages_for_feature(
         gate.put()
 
 
-
 class FeatureEditHandler(basehandlers.FlaskHandler):
 
+  # Field name, data type
+  EXISTING_FIELDS = [
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply pages/guide.py]: Added 🙂

File: pages/guide.py [hunk 4]
```diff
@@ -148,19 +235,42 @@ def touched(self, param_name: str) -> bool:
...
+    elif field_type == 'str':
+      return self.form.get(field)
+    elif field_type == 'split_str':
+      val = self.split_input(field, delim=',')
+      if field == 'blink_components' and len(val) == 0:
+        return [settings.DEFAULT_COMPONENT]
+      return val
+    elif field_type == 'bool':
+      return self.form.get(field) == 'on'
+    elif field_type == 'special':
```
[the remark this answers is not in the document]
Comment (DanielRyanSmith) [reply pages/guide.py]: A relic from a previous implementation - now removed!

**Review by jrobbins — APPROVED** [review]

**Pull request merged**